src/Fixer/ClassNotation/MethodSeparationFixer.php

```diff
@@ -76,159 +113,187 @@ public function getPriority()
     /**
      * {@inheritdoc}
      */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAnyTokenKindsFound(Token::getClassyTokenKinds());
+    }
+
+    /**
+     * {@inheritdoc}
+     */
     protected function applyFix(SplFileInfo $file, Tokens $tokens)
     {
         $tokensAnalyzer = new TokensAnalyzer($tokens);
+        $class = $classStart = $classEnd = false;
```

Comment:
what is `$class` here if not start nor end ?

Reply:
`$class` value doesn't matter here, it is to keep SCA happy. Whenever `$class` is used as value it is the index of a classy (`T_CLASS`, `T_INTERFACE`, `T_TRAIT`). It cannot be otherwise unless invalid Tokens collection is used (i.e. tokenized invalid PHP).

Reply:
then what's the difference between `$class` and `$classStart` ?

Reply:
>then what's the difference between $class and $classStart ?

`$class` is provided by the analyzer, `$classStart` is the index of the opening `{` of the class.